modules/service/src/main/scala/lucuma/odb/logic/Generator.scala

```diff
@@ -408,18 +419,17 @@ object Generator {
           .fromEither(Error.sequenceTooLong.asLeft[ExecutionDigest])
           .unlessA(ctx.scienceIntegrationTime.toOption.forall(_.exposureCount.value <= SequenceAtomLimit)) *>
         (ctx.params match
+          case GeneratorParams(_, _, config: f2.longslit.Config, role, declaredComplete, _) =>
+            flamingos2LongSlit(ctx, config, role, when).flatMap: (p, e) =>
+              EitherT.fromEither[F](executionDigest(p, e, calculator.flamingos2.estimateSetup))
+
           case GeneratorParams(_, _, config: gmos.longslit.Config.GmosNorth, role, declaredComplete, _) =>
             gmosNorthLongSlit(ctx, config, role, when).flatMap: (p, e) =>
               EitherT.fromEither[F](executionDigest(p, e, calculator.gmosNorth.estimateSetup))
 
           case GeneratorParams(_, _, config: gmos.longslit.Config.GmosSouth, role, declaredComplete, _) =>
             gmosSouthLongSlit(ctx, config, role, when).flatMap: (p, e) =>
               EitherT.fromEither[F](executionDigest(p, e, calculator.gmosSouth.estimateSetup))
-
-          case GeneratorParams(_, _, config: f2.longslit.Config, _, _, _) =>
-            EitherT.leftT[F, ExecutionDigest](
-              Error.InvalidData(ctx.oid, "F2 longslit not implemented")
```

Comment:
I just pushed this into a `flamingos2Longslit` method for now.  It is shared by the digest calculation and the sequence generator.